### PR TITLE
googletest module

### PR DESCRIPTION
**Pull request by saneck2150**

### CODE REVIEW

File: cmake/ThirdParty.cmake [hunk 1]
```diff
@@ -180,6 +180,31 @@ include_directories(SYSTEM
   ${CMAKE_CURRENT_BINARY_DIR}/3rdparty/cli11-prefix/include
 )
 
+#google test
+include(ExternalProject)
```
Comment (gavv) [inline cmake/ThirdParty.cmake:184]: This include is already present at the top of the file.

File: cmake/ThirdParty.cmake [hunk 2]
```diff
@@ -180,6 +180,31 @@ include_directories(SYSTEM
...
+
+ExternalProject_Add(
+    googletest
+    SOURCE_DIR "${CMAKE_BINARY_DIR}/googletest-src"
+    BINARY_DIR "${CMAKE_BINARY_DIR}/googletest-build"
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG release-1.11.0
+    GIT_SHALLOW ON
```
Comment (gavv) [inline cmake/ThirdParty.cmake:192]: If we're building from submodule, these options shouldn't be needed.

File: cmake/ThirdParty.cmake [hunk 3]
```diff
@@ -180,6 +180,31 @@ include_directories(SYSTEM
...
+    INSTALL_COMMAND ""
```
Comment (gavv) [inline cmake/ThirdParty.cmake:193]: Please take a look how other dependencies are added in ThirdParty.cmake.

We install each dependency to a prefix dir, then add prefix dir to include path and add specific libs using link_libraries().

We also use CMAKE_ARGS to pass all necessary cmake options to allow cross-compilation.

You also need to add gtest to ALL_DEPENDENCIES list.

File: cmake/ThirdParty.cmake [hunk 4]
```diff
@@ -180,6 +180,31 @@ include_directories(SYSTEM
...
+    TEST_COMMAND ""
+    LOG_DOWNLOAD ON
+    LOG_BUILD ON
+)
+
+option(BUILD_TESTING "Build tests" ON)
```
Comment (gavv) [inline cmake/ThirdParty.cmake:199]: Other options are defined in root CMakeLists.txt, so it's better to move it there too.

**Comment by saneck2150** [comment]
Google tests module

File: cmake/ThirdParty.cmake [hunk 5]
```diff
@@ -180,6 +180,31 @@ include_directories(SYSTEM
...
+
+if (BUILD_TESTING)
```
Comment (gavv) [inline cmake/ThirdParty.cmake:201]: If BUILD_TESTING is OFF, we should not build gtest at all.

**Review by gavv — COMMENTED** [review]
Thanks for PR

**Pull request closed without merging**